### PR TITLE
Fix urllib3.util.Retry.sleep type

### PR DESCRIPTION
**Pull request by philipdexter**
https://github.com/urllib3/urllib3/blob/main/src/urllib3/util/retry.py#L342

### CODE REVIEW

File: stubs/urllib3/urllib3/util/retry.pyi [hunk 1]
```diff
@@ -43,7 +43,7 @@ class Retry:
     @classmethod
     def from_int(cls, retries, redirect=..., default=...): ...
     def get_backoff_time(self): ...
-    def sleep(self): ...
+    def sleep(self, response=...): ...
```
Comment (srittau) [inline stubs/urllib3/urllib3/util/retry.pyi:46]:
```suggestion
    def sleep(self, response: HTTPResponse | None = ...) -> None: ...
```

Also needs an import from `..response`.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks. I suggest to properly annotate it, while we're here. You also need to remove the following entries from `stubs/urllib3/@tests/stubtest_allowlist.txt` to make the tests pass.

* `urllib3.Retry.sleep`
* `urllib3.connectionpool.Retry.sleep`
* `urllib3.util.Retry.sleep`
* `urllib3.util.retry.Retry.sleep`

File: stubs/urllib3/urllib3/util/retry.pyi [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from typing import Any
 
 from .. import exceptions
+from http.client import HTTPResponse
```
Comment (srittau) [inline stubs/urllib3/urllib3/util/retry.pyi]: This is a different class:

```suggestion
from ..response import HTTPResponse
```

**Review by srittau — APPROVED** [review]

**Comment by philipdexter** [comment]
Thanks! I think I got all your changes in now

**Pull request merged**